dunamai/__init__.py

```diff
@@ -281,7 +301,11 @@ def serialize(
             style = Style.Pep440
         out = ""
 
-        meta_parts = []
+        # treat local segments as the first meta_part
+        if self.local:
+            meta_parts = [self.local]
+        else:
+            meta_parts = []
         if metadata is not False:
```

Comment:
The only potential issue I see with enabling this by default, in terms of backwards compatibility, is that someone currently could have tags like `v1.2.3+foo_bar`, which doesn't hurt anything today, but would start to produce an error because PEP 440 does not allow `_` in the local segment.

To be safe, let's make this opt-in:

* Add a new boolean argument to this function (`tagged_metadata: bool = False`)
* Add a corresponding CLI option (`--tagged-metadata`)
* In my mind, `--tagged-metadata` would work like `--dirty` in that it's extra, optional metadata, so I think it should be subject to the `--no-metadata` flag just like `--dirty` is. That just means you would put it inside of the `if metadata is not False:` condition here.